### ISSUE TITLE
`t` doesn't mark TODO in org-mode

### ISSUE THREAD

**Issue by JuanCaicedo** [syl20bnr/spacemacs#8640]
#### Description :octocat:

`t` in org-mode (or evil-org-mode) no longer marks header as TODO. Not sure if
this is an upstream problem, feel free to close if so.

#### Reproduction guide :beetle:
- Start Emacs
- Open org-mode
- Make header like 
```org
* test
```
- Press `t`

*Observed behaviour:* :eyes: :broken_heart:
- in messages buffer
    > user-error: Can’t find t

*Expected behaviour:* :heart: :smile:
- Line should be marked as `TODO`

#### System Info :computer:
- OS: darwin
- Emacs: 25.1.1
- Spacemacs: 0.200.8
- Spacemacs branch: develop (rev. f79a169)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(nginx csv helm auto-completion command-log emacs-lisp elm extra-langs git github html javascript ibuffer markdown
       (org :variables org-enable-github-support t)
       osx php python ruby yaml shell-scripts themes-megapack better-defaults spell-checking
       (syntax-checking :variables syntax-checking-enable-by-default nil)
       version-control)
```
- System configuration features: JPEG RSVG NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES

#### Backtrace :paw_prints:
```
<<BACKTRACE IF RELEVANT>>
```

**Comment by Biotechjones** [comment]
I get similar issues. Its as if the org keycap doesn't get loaded when you load an org-file

**Comment by ngoonee** [comment]
Having the same thing on master (not develop). Manually loading the org-mode major mode doesn't help (it seems to load correctly anyway)

**Comment by Voleking** [comment]
I get similar issues. And I wander if there is any way to fix this issue temporarily.

**Comment by Voleking** [comment]
It seems that this [line](https://github.com/syl20bnr/spacemacs/blob/develop/layers/+emacs/org/packages.el#L56) should be 
`(setq evil-org-key-theme '(textobjects navigation additional insert todo))`
now `t` mark TODO in org-mode.

**Comment by TheBB** [comment]
Thanks for bringing this to my attention. In develop, `t` no longer marks todo by default. As you say, you can configure that variable or use `SPC m TT` instead. Since this is a non-issue I'll put this as fixed in develop.

**Comment by kccai** [comment]
I think this is a bug since `t` is documented as `org-todo` in the latest org layer readme.org.


https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Bemacs/org/README.org

**Comment by TheBB** [comment]
Thanks, somehow I didn't expect to find todo bindings under “tree”. So many bindings there that don't belong. Removed it.

**Comment by Voleking** [comment]
@TheBB as for me, `org-todo` is nearly the function I call most. So keep this kind of shortcut as default will be nice.

**Comment by TheBB** [comment]
As you said it's easy to customize though. I find it difficult to believe that so many people need to do this so often that we have to have a **one key** shortcut for it, but I'm open to listening.

**Comment by JuanCaicedo** [comment]
I agree that I use it hundreds of times daily. I probably do that just since it was included in spacemacs when I started using it, but I find it incredibly useful. 

I find it to be one of those things that I didn't know I was missing until I had it. A user new to emacs might not know how to use `org-todo` as a function, much less how to map it to their own keybinding (I definitely wouldn't have). 

What's the value in removing this keybinding? Can we trace back to who added it originally and figure out what value they saw in adding it?  

I don't want to bike shed this if you feel it's not important. It's just that to me it's a really useful feature and I think others might like it and not know it's going away 😃

**Comment by TheBB** [comment]
100 times daily corresponds to once every six minutes, assuming you're at the computer for 10 hours. If that is true you are probably far north of the average user, let alone hundred**s** of times daily.

I can add a layer variable though if you feel so strongly about it.

**Comment by TheBB** [comment]
https://github.com/syl20bnr/spacemacs/commit/010164ce778098b309fcd1cbd99f42b5c4ba5b5c

**Comment by Somelauw** [comment]
Would it be a good idea to bind C-t to org-todo in Spacemacs?

By the way, each of the following key combinations can also be used to change the TODO status:
- SPC m L
- C-c C-t
- S-right

**Comment by kepi** [comment]
First of all, I'm trying to be constructive. If anything in this post seams offensive, it wasn't intended (blame my english).

Thanks for layer variable. It solves problem for me because I spent hour and half finding where the problem is and then going through commits and issues, solving it myself, reporting and then finding there is the solution. Unfortunately I think there are many ppl for which this will remain problem because they are not able to find the solution.

I suppose this is not right place to discuss this but main problem is that bindings shouldn't be changed at all if possible. And definitely not as often as in org layer in last months.

It is extremely inconvenient for users. I think I can understand most reasons to change something, but this is ... saying politely... "not smart". I can accept that something breaks - it comes with using develop. But then I realize that somebody break smart shortcut intentionally and that is something I don't think we should accept.

Please suggest where it is best place to discuss/solve this (again, probably not in this issue) but at least think about:

- shortcuts are often in one's muscle memory. Someone's hurried decision shouldn't complicate life of dozen's or hundreds ppl
- shortcuts should be decided at start of project. So it is ok to choose whatever shortcuts you want when you are starting new layer
- shortcuts shouldn't be changed if not absolutely needed. Definitely not because somebody wasn't thinking them enough at start
- changing shortcuts that people use a lot from one letter to 3 letters is not smart
- adding new shortcut so people can discover it is smart - but leave original at place if possible

And *most importantly* please if there is a change:

1. **discuss those changes in advance** - but it wouldn't help a lot as most people will not know about discussion (maybe if combined with 4.)
2. make commits with **clear subjects** from which we can learn that something changed
3. describe in commits **how we can revert** that change
4. we should have some changes alerts on home spacemacs buffer. It would be much better if we see "Hey, there is shortcut change" after spacemacs update. Even in develop.

It is really totally insane to introduce such change in commit with subject "Use evil-org from MELPA".

btw. C-t doesn't seam good idea to me as I'm evil user. S-right is for cycling through states, it doesn't offer popup with available states.

**Comment by jb55** [comment]
Marking todo DONE with 3-4 keystrokes is insane. This was clearly done by someone who doesn't use org-mode a lot??? *turns on option* good day.

**Comment by Tiberriver256** [comment]
Definitely makes it a bit more difficult to use. I just started using emacs just for org-mode and I think there are a lot of others out there like myself, org-mode seems to be the gateway drug.

**Comment by PrestanceDesign** [comment]
@Tiberriver256 There is the option **org-want-todo-bindings** to retrieve simplicity ;)
https://github.com/syl20bnr/spacemacs/commit/010164ce778098b309fcd1cbd99f42b5c4ba5b5c

**Comment by adamczykm** [comment]
Bumping this topic. I'm just another 'used-to-be-happy-with-former-t-in-org-mode' guy. If you truly use this mode for organising your agendas you will need marking subtrees with TODO statuses to be as efficient as possible.

**Comment by nixmaniack** [comment]
@adamczykm Could you tell more about the issue you're facing? This has been fixed in develop and you can enable the old functionality of marking todo states with `t` with  `org-want-todo-bindings` as mentioned in the comments above.

**Comment by pangwa** [comment]
It's not a good user experience since the default behavior was changed! It doesn't make any sense for the user to adjust the config to bring back the old (and useful) behavior. 

I spent ~15 minutes to fix my config(set org-want-todo-bindings t) and I think many org users may waste their time doing the same fix...

**Comment by cyinwei** [comment]
For reference, here's how to enable it from [liujoey](https://github.com/syl20bnr/spacemacs/issues/10495#issuecomment-374763113).

```
dotspacemacs-configuration-layers
   '(
     (org :variables org-want-todo-bindings t)
)
```

Complete newbie at elisp and Spacemacs, and that comment helped a lot.

**Comment by gnohz** [comment]
@cyinwei This works, thanks!

I just switched to develop branch, and some commonly used keybindings are changed in orgmode, in addition to this one.

For example, to activate a pomodoro, I used to press Spc m p, now you have to press Spc m C p.
But this is not as documented as in the org-layer readme, which makes things very inconsistent.

Also, when you cancel a pomodoro, it used to clock out and stop the current one. But now it will cancel the pomorodo and delete the current clock. I think there is a way to make things to work out, but why not keep the most popular package consistent across different branches?

Just my two cents.

**Comment by Miciah** [comment]
> Also, when you cancel a pomodoro, it used to clock out and stop the current one. But now it will cancel the pomorodo and delete the current clock. I think there is a way to make things to work out, but why not keep the most popular package consistent across different branches?

I don't think the master and develop branch are different in this respect.  Is it possible you had an older version of the org-pomodoro package installed, and the package got updated when you changed Spacemacs branches?

Anyway, I think you want to following setting:
```emacs-lisp
(setq org-pomodoro-keep-killed-pomodoro-time t)
```

**Comment by gnohz** [comment]
@Miciah I installed the develop branch from scratch, so I don't think the package is obsolete. Even though, I deleted "org layer", and put is back, I have the version of org-pomodoro as before: org-pomodoro-20161119.226

If this is not the updated version, could you please let me know how to update it to the newest version?

Thanks.

**Comment by gnohz** [comment]
I used spacelpa, so I have older versions of packages.
Now I make the spacelpa nil, and I have the same version of org-pomodoro as yours. However, it does not solve the problem, it still deletes the time stamps.

**Comment by Miciah** [comment]
Did you try the `org-pomodoro-keep-killed-pomodoro-time` setting?

**Comment by gnohz** [comment]
@Miciah Yes, this works! Thank you very much. By the way, do you know how to change the keybinding Spc m C p to Spc m p to start a pomodoro?

**Comment by Miciah** [comment]
@gnohz, try the following:
```emacs-lisp
(spacemacs/set-leader-keys-for-major-mode 'org-mode "p" 'org-pomodoro)
```

**Comment by Miciah** [comment]
The following will remove the old binding:
```emacs-lisp
(spacemacs/set-leader-keys-for-major-mode 'org-mode "Cp" nil)
```

**Comment by gnohz** [comment]
@Miciah unfortunately, these don't work...

**Comment by ret2libc** [comment]
I agree this change was a bit of a surprise. I had to spend time just to enable an option to bring things back to what they were. In org mode I often move items between different states, so using `t d` or `t g`, etc. was really great.

**Comment by IngvarListard** [comment]
Since I upgrade spacemacs to developer version 't' key doesn't work at all even with variable set.

**Comment by windedge** [comment]
In my case, it was caused by ipython-notebook layer, I commented out ipython-notebook layer, and there was no problem anymore.

**Comment by Asalle** [comment]
Please don't change the default keybindings. I understand you're the maintainer and it's up to you, but as a user I'm telling you: it's killing the magic of spacemacs for me.

**Comment by pbolger** [comment]
> As you said it's easy to customize though. I find it difficult to believe that so many people need to do this so often that we have to have a **one key** shortcut for it, but I'm open to listening.

Excuse me for sounding a bit peeved, but I've just spent a couple of weeks wondering why Org mode in Spacemacs doesn't work properly. 

Cycle TODO is listed in [Org Mode Basics](https://orgmode.org/manual/TODO-basics.html)  as the first keyboard shortcut. 

> The most important commands to work with TODO entries are:
> 
> C-c C-t     (org-todo)
> Rotate the TODO state of the current item 

It also is one of only 15 one key shortcuts in the [Spacemacs manual](http://spacemacs.org/layers/+emacs/org/README.html#key-bindings). 

You'll also find that it's referred to in countless third party "Getting started with Org Mode in Spacemacs" videos. 

So why would you remove a key feature, and seemingly on a whim? 

This looks like a mistake, particularly when it seems that the "T"  (org-insert-todo-heading nil) to make a new heading shortcut seems to have disappeared as well.

**Comment by duianto** [comment]
This seems to have been why it was changed:
>My reasoning is
>
>* single-shortcut keys are few in number and valuable
> * the vim/evil defaults for those keys are designed for editable modes, and org is an editable mode
> * practically nowhere else in Spacemacs are such keys overridden
> * the benefit of making an exception in this case is dubious
> * it's not too difficult to configure on a user level
>
>For sure, many people will be toggling todo state more often than using the `t` jump operator while in org-mode, but I'm not convinced that this is worth violating POLA in org-mode. But of course if there is consensus that I'm wrong then I'm happy to be convinced otherwise.


source: Use evil-org from MELPA https://github.com/syl20bnr/spacemacs/pull/9041#issuecomment-309758098

**Comment by pbolger** [comment]
> * the vim/evil defaults for those keys are designed for editable modes, and org is an editable mode

I can see the merit in this argument, but surely demoting it to three levels deep  - and without changing the docs to alert develop branch users - is pretty unhelpful. Particularly when there are so many resources out there that refer to the original behaviour.

Going back to the Org Mode documentation the developers there obviously regard 'Make heading', 'Cycle TODO status' , 'promote/demote' and 'move' as core Org Mode functions. IMO Spacemacs should treat them as such.

As 't' and 'T' both have underlying Vim editing functions they are probably not a good choice but "SPC m TT" is agonisingly slow for something which should be as easy as switching between 'normal' and 'insert'.

The alternatives all have their problems too:
SPC m L -- Still too many keypresses for what should be a toggle.
C-c C-t  -- If I'm going to have to use Holy mode key combinations, what am I doing in Evil Mode*?
S-right -- Forces you to leave the keyboard, unless you are using some odd keyboard with the arrow keys on a separate layer. 

* Went through the stock Emacs tutorial yesterday, and _it wasn't that bad!_

**Comment by Asalle** [comment]
I don't know if it's how it's usually done, but:
it's a UX question, so let's have a poll or interview some users and see what they have to say about the change?

**Comment by Apromixately** [comment]
> Would it be a good idea to bind C-t to org-todo in Spacemacs?
> 
> By the way, each of the following key combinations can also be used to change the TODO status:
> 
>     * SPC m L
> 
>     * C-c C-t
> 
>     * S-right

I think there is an important issue buried here.
Checking `h d f org-todo`:
```
org-todo is an interactive compiled Lisp function in ‘org.el’.

It is bound to M-RET T T, M-m m T T, C-c C-t, and many ordinary text characters.
```
and with `f1 w org-todo` the list becomes much longer but both `S-right` and `, L` are *NOT* on that list. These are both bound to `org-shiftright` which apparently also toggles todo states.
I don't want to say that this is spacemacs problem, necessarily. It might be considered an org-mode problem but I think it's quite unfortunate when such shortcuts are not discoverable.

**Comment by guysbryant** [comment]
I was watching a tutorial this morning in which the author demonstrated that `t` would cycle the todo statuses of a tree element. Thankfully he also showed that `S-arrow keys` would cycle them as well because the documentation only says `t` for this. After reading through this thread I learned that `SPC m T T` would do it as well as `SPC m L`. 

Keeping in mind that `,` is the same thing as `SPC m`, and that most of us are using evil mode and have the power of the dot at our disposal, this doesn't really seem too bad. My workflow is to use `, L` to do the first toggle of a todo status, then to use the dot `.` to do any further cycling. I think this is a fine method for handling this toggle and we get to keep `t` for jumping which is a small, but nice bonus.

tl;dr
`, L` to toggle once then use `.` for further toggling and you get to keep the vim `t` jump command.

**Comment by blackdev1l** [comment]
This issue is first link when you google "org mode can't find t todo" so i'm going to write here: 
Is there an easy way to get the shortcut back? it's probably the most used one in org-mode and it's not going to change.

**Comment by tshu-w** [comment]
@blackdev1l `(setq org-want-todo-bindings t)`

**Comment by micahbf** [comment]
Chalk me up as another person who spent an hour figuring out why this binding went away.

**Comment by bderrly** [comment]
> @blackdev1l `(setq org-want-todo-bindings t)`

I discovered that I need to add this under `dotspacemacs/user-init()` rather than `dotspacemacs/user-config()`

**Comment by kepi** [comment]
> > @blackdev1l `(setq org-want-todo-bindings t)`
> 
> I discovered that I need to add this under `dotspacemacs/user-init()` rather than `dotspacemacs/user-config()`

You can set it directly when setting layer. Instead of adding only

```elisp
org
```

to `dotspacemacs-configuration-layers` add it as:

```elisp
     (org :variables
          org-want-todo-bindings t)
```

**Comment by blackdev1l** [comment]
> > @blackdev1l `(setq org-want-todo-bindings t)`
> 
> I discovered that I need to add this under `dotspacemacs/user-init()` rather than `dotspacemacs/user-config()`

Thanks for the clarification, i'm going to look into it as soon as possible!

**Comment by camio** [comment]
Hi, new user here. I'm learning both spacemacs and org mode and figured there's got to be a better way than ', T T'. I appreciate that 'org-want-todo-bindings' provides me a way to get the one-key behavior, but I am certain that the current default (org-want-todo-bindings f) is wrong.

To see why, we need to look at the principals of spacemacs (mnemonic based editing) and org mode (where TODOs are a primary purpose for its existence). I appreciate that binding 't' can seem like a binding land grab, but in this case it is justified. Org mode is going to be something that spacemacs users will want to try very quickly and if it doesn't have a short binding for such a common operation, then I think it'll hurt adoption.

Here's another way of looking at it, if (org-want-todo-bindings t) is the default, how many will switch it to (org-want-todo-bindings f)? I would venture to guess many fewer people care about having the vim 't' binding in org mode than those who care about having a fast TODO marker in org mode. Lets not let ideology get in the way of pragmatism here.

**Comment by GaneshParkopedia** [comment]
Is spacemacs a dead project? As a new user (long time emacs user), I followed the installation instructions for the plain spacemacs installation. I read the [org mode readme page](https://www.spacemacs.org/layers/+emacs/org/README.html), and found out that my keyboard shortcuts don't work. This has lead me to a 4 year old post that says that the documentation is out of date.

Whether or not the (not very) new key binding is a good idea, I think docs should be updated when changing them, otherwise it is very confusing for new users. Considering the amount of backlash about this particular change, mentioning the flag in the readme to be able to use those shortcuts would be pretty helpful

**Comment by Miciah** [comment]
> Is spacemacs a dead project?

There hasn't been a release in quite some time, but the develop branch is fairly lively.  See https://develop.spacemacs.org/ the develop branch's documentation; https://develop.spacemacs.org/layers/+emacs/org/README.html#org-with-evil-org-mode documents the `org-want-todo-bindings` option.

**Comment by GaneshParkopedia** [comment]
> There hasn't been a release in quite some time, but the develop branch is fairly lively. See https://develop.spacemacs.org/ the develop branch's documentation; https://develop.spacemacs.org/layers/+emacs/org/README.html#org-with-evil-org-mode documents the `org-want-todo-bindings` option.

Ah I understand now, I didn't realise I should be using the develop documentation page. Thanks!

**Comment by lebensterben** [comment]
@shiki 
cannot reproduce.
When you move the cursor onto a heading in org-mode, and evaluate (describe-key "t"), do you see `org-todo`?

**Comment by lebensterben** [comment]
This doesn't make any sense.

In your config you've set `org-want-todo-bindings` to t. https://github.com/shiki/dotfiles/blob/ae4235e4fe6449f27fec4bdc71640bbf586f94de/spacemacs/.spacemacs#L58

So [this](https://github.com/syl20bnr/spacemacs/blob/5c88c3b089c132997bf9f571a0207d93739efef1/layers/%2Bemacs/org/packages.el#L92-L95) evaluates to `(setq evil-org-key-theme '(textobjects navigation additional todo))`. (You can verify it by `(describe-variable 'evil-org-key-theme)`).

When `evil-org` is loaded, ]this line](https://github.com/Somelauw/evil-org-mode/blob/80ef38fb378541937f6ddfe836809e76eda1e355/evil-org.el#L816) is always evaluated. It finds `todo` in `evil-org-key-theme` and therefore setup todo bindings. https://github.com/Somelauw/evil-org-mode/blob/80ef38fb378541937f6ddfe836809e76eda1e355/evil-org.el#L794

Specifically, https://github.com/Somelauw/evil-org-mode/blob/80ef38fb378541937f6ddfe836809e76eda1e355/evil-org.el#L731-L732 is equivalent to `(define-key evil-normal-state-map (kbd "t") 'org-todo)`

**Comment by xabbu42** [comment]
Just wanted to note that having a complete environment where I can assume vim keybindings everywhere is by far the biggest selling point of spacemacs. I use t in vim a lot and would have been disappointed if it had not worked in an org buffer.
